app/models/presenter/auction.rb

```diff
@@ -94,8 +94,46 @@ def html_summary
       markdown.render(summary)
     end
 
+    def status
+      if available?
+        'Open'
+      else
+        'Closed'
+      end
+    end
+
+
+    delegate :label_class, :label, :tag_data_value_status, :tag_data_label_2, :tag_data_value_2,
+      to: :status_presenter
+
+    def human_start_time
+      if start_datetime < Time.now
+        # this method comes from the included date helpers
+        "#{distance_of_time_in_words(Time.now, start_datetime)} ago"
+      else
+        "in #{distance_of_time_in_words(Time.now, start_datetime)}"
+      end
+    end
+
     private
 
+    def status_presenter_class
+      status_name = if expiring?
+                      'Expiring'
+                    elsif over?
+                      'Over'
+                    elsif future?
+                      'Future'
+                    else
+                      'Open'
+                    end
+      "::Presenter::AuctionStatus::#{status_name}".constantize
```

Comment:
So the pattern is called something like 'replace conditional with polymorphism'. http://refactoring.com/catalog/replaceConditionalWithPolymorphism.html

The goal is to replace the same conditional in a bunch of places into some fairly stupid declarative objects ... and only have the conditional in one place. 

Sandi Metz did a talk at Rails conf called (I think) "Nothing is something" that takes this a bit further and tries to find the missing thing that is covered by nil or false.